compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -613,6 +724,27 @@ impl<'a> Parser<'a> {
 
                     self.parse_pat_range_begin_with(begin.clone(), form)?
                 }
+                // HACK: `self.maybe_recover_trailing_expr` will recover the pat to `_` if the pat isn't valid,
+                // so let's just check if `_..pat` is valid
+                // the `pat-recover-wildcards` test will check that `_..pat` won't actually compile
```

Comment:
Instead of (ab-)using `PatKind::Wild`, you could experiment with creating & using `PatKind::Err` isntead (similar to `ExprKind::Err`) with which you might be able to resolve this HACK? Dunno, I've only *skimmed* this PR so far.

If you do, please consider creating a separate commit for it since you will need to modify a bunch of crates `rustc_{ast,ast_pretty,expand}` and likely some more.

Reply:
And while `ExprKind::Err` is a unit variant, ideally it should be `Err(ErrorGuaranteed)` which it isn't for historical reasons (I don't know if there are any blockers which would prevent updating it).

All that to say, in my opinion the hypothetical `PatKind::Err` should take `ErrorGuaranteed` as an argument if possible.